cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -288,12 +303,31 @@ func deviceLayoutFromPartitionTable(ptable sfdiskPartitionTable) (*DeviceLayout,
 		}
 	}
 
+	// if we don't have an LBA, we are probably on an MBR disk and need to
+	// calculate the size ourselves
+	// TODO:UC20: how/where/why/who/when to best do this ???
+	var numSectors gadget.Size
+	if ptable.LastLBA != 0 {
+		numSectors = gadget.Size(ptable.LastLBA + 1)
+	} else {
+		out, err := exec.Command("blockdev", "--getsz", ptable.Device).CombinedOutput()
```

Comment:
Replaced with reading the sector size directly from `/sys/block/$(basename dev)/size`

Reply:
Actually rolled that back, and just wrapped the call to blockdev nicely.